### DISCUSSION TITLE
Weather forecast plot

### DISCUSSION THREAD

**Discussion by 123-fake-st** [has2k1/plotnine#834] [2024 Plotnine Contest]
### Authors

123-fake-st

### Links

[github](https://github.com/123-fake-st) | [LinkedIn](https://www.linkedin.com/in/john-t-04b9578)

### Full description

# Weather forecast plot

A plot showing a 7-day weather forecast.

![weather forecast](https://github.com/has2k1/plotnine/assets/34491334/41c84fa4-c242-4315-bced-7b45d9293bc6)

View the [jupyter notebook](https://github.com/123-fake-st/2024_plotnine_contest/blob/main/weather_forecast.ipynb) for the code and a walkthrough of generating this plot including several prototypes working up to the final plot.

The code uses plotnine's `geom_path`, `geom_label`, and `watermark` to build this plot.
Two `geom_path`s are layered for the temperature ranges.
`scale` and `theme` settings polish the plot's look.

## Motivation

While working out what to do in this project I was motivated by these factors:

1. Provide visual demonstration of something where there isn't a visual example in the plotnine documentation's [reference](https://plotnine.org/reference/), [gallery](https://plotnine.org/gallery/), or [tutorials](https://plotnine.org/tutorials/) pages
1. Make something beautiful
1. Make something that I didn't know how to make already
1. Explain it well

The data for this plot is quite simple and is very different from the large data sets I work with on a typical day.
However, creating the polished plot has some interesting challenges and I hope that even many experienced plotnine users will see this plot, wonder how it was made, and want to check out the code to understand it.

## Other notes

I placed slightly tweaked code from the jupyter notebook into a python function that you can import and call to generate plots by providing your own forecast data.

```python
from plot_forecast import plot_forecast
plot_forecast('./data/Madison.csv', location='Madison, WI', outfile='Madison.png')
```

### Code repository

[repo](https://github.com/123-fake-st/2024_plotnine_contest) | [jupyter notebook](https://github.com/123-fake-st/2024_plotnine_contest/blob/main/weather_forecast.ipynb) | [README](https://github.com/123-fake-st/2024_plotnine_contest/blob/be27a533d1521131e318188453365434df38362a/README.md)

**Comment by kyleGrealis** [comment]
this really looks amazing and your `.ipynb` is well-documented!

**Reply by 123-fake-st** [reply]
Thank you!  I genuinely appreciate the feedback: most of my work doesn't get to go out in public and I was waffling about putting in a submission; I'm glad that I entered something.

**Reply by kyleGrealis** [reply]
Nah it's great! And I picked up a few things. I'll be finalizing my submission by early next week. I'm motivated to enter after seeing the good work

**Comment by Ringomed** [comment]
Very pretty visualization!

**Comment by has2k1** [comment]
![weather](https://github.com/user-attachments/assets/62740194-c42d-403f-b681-697c01887550)

**Comment by EconomiCurtis** [comment]
Ahoy @123-fake-st , 
I am one of the contest organizers. 
If you'd like, we'd be happy to send you a little Plotnine and Posit care package of hex-shaped stickers. Just a small thanks for taking part. https://forms.gle/M8QmM286MSHbkDRd7

**Reply by 123-fake-st** [reply]
Thanks @EconomiCurtis!  It appears that this link didn't get commented onto the submissions that are past page 1.  Please don't forget our colleagues on the other pages 😄

**Comment by Lerac1275** [comment]
This is so well done. Amazing!

**Comment by has2k1** [comment]
@123-fake-st, The winning submission has been announced [here](https://posit.co/blog/winner-of-the-2024-plotnine-plotting-contest/). Thank you for taking part in the contest.

**Reply by 123-fake-st** [reply]
@has2k1 thank you so much for your work on plotnine and hosting the contest.  There were many great submissions and I learned a lot.
